src/hooks/web3/index.tsx

```diff
@@ -107,11 +109,12 @@ export function Provider({
   )
 }
 
-const onError = (error: Error) => console.error(error)
+const onErrorFallback = (error: Error) => console.error(error)
 
 function initializeWeb3ReactConnector<T extends Connector, P extends object>(
   Constructor: { new (options: P): T },
-  options?: Omit<P, 'actions'>
+  options?: Omit<P, 'actions'>,
+  onError: (error: Error) => void = onErrorFallback
```

Comment:
This does not need a fallback, and it is already included in the options pack, so it is confusing to additionally include it separately. You can remove `onError` and `onErrorFallback` entirely.

Instead, explicitly pass `onError: console.error` in the options pack in all invocations of `initializeWeb3ReactConnector`. If you want to keep it as a fallback, then add something like this to `initializeWeb3ReactConnector:

      options: Omit<P, 'actions'>, // makes options required (this should probably be done anyway)
    ...
      options.onError = options.onError ?? console.error